frame/offences/src/tests.rs

```diff
@@ -71,14 +75,34 @@ fn should_not_report_the_same_authority_twice_in_the_same_slot() {
 	});
 }
 
+#[test]
+fn should_not_report_an_obsolete_offence() {
+	new_test_ext().execute_with(|| {
+		// given
+		let time_slot = 42;
+		let session_index = 3;
+		assert_eq!(offence_reports(KIND, time_slot), vec![]);
+
+		let offence =
+			Offence { validator_set_count: 5, session_index, time_slot, offenders: vec![5] };
+		assert_eq!(Offences::report_offence(vec![], offence), Err(OffenceError::ObsoleteReport));
```

Comment:
```suggestion
		assert_noop!(Offences::report_offence(vec![], offence), OffenceError::ObsoleteReport);
```

With transactional-by-default, this is less of a concern, but still a good practice.

macro ensures that no storage has changed within the call.